### DISCUSSION TITLE
A new method to remove Agent from the cloud

### DISCUSSION THREAD

**Discussion by tkatsoulas** [netdata/netdata#13163] [Ideas]
I would like to suggest a new way to remove the Agent from the cloud. This method could be integrated in the `netadacli`. In this approach the Agent will ask from the cloud to remove it from a space

This seems like a simple feature to implement, want we will need:  
1. A request to the cloud to remove this Agent from the space, mark it's uuid us unclaimed
2. local changes in the Agent to remove any cloud claim configuration.

Rational behind that: 

Ephemeral node support in Netdata Cloud. ASGs can have a launch template script that installs the agent and claims it in a Netdata cloud space but when there is a scale down you have a bunch of servers hanging as not reachable in Netdata cloud dashboard. Leaving here this (https://aws.amazon.com/blogs/infrastructure-and-automation/run-code-before-terminating-an-ec2-auto-scaling-instance/) document which describes how you can make instructions on executing a script during shutdown on AWS.

**Comment by shyamvalsan** [comment]
@tkatsoulas ephemeral node support is on the todo list, the idea is that the user would provide during install time an indication that this node (or node instance) is ephemeral and such nodes would automatically get cleaned up instead of the user having to clean it up manually in Netdata cloud. Does this solve what you are looking for?

**Comment by andrewm4894** [comment]
I feel like there just should be a set of rules or filters in cloud that just move nodes to trash. 

That way you could just have one run to clean up all nodes tagged as ephemeral in whatever way you want.

And lots of other scenarios too.

We should just build layer to handle the logic for when nodes go to trash. Give sensible default rules that users can enable. But also ability for more flexible approach if they want.
